### ISSUE TITLE
Chat based commands not working in 1.18.2 forge

### ISSUE THREAD

**Issue by ShinQdan** [CraftTweaker/CraftTweaker#1581]
### Issue description

I have CraftTweaker installed through CourseForge desktop app. When using 1.18.2 version (basically any, tested on newest and some older) on Forge (also tested some versions of it) it does not display any chat info when using commands - /ct hand, inventory, log, discord, scripts, issues. The examples and recipes commands work but do not display any message on chat.

### Steps to reproduce

1. Install CourseForge app.
2. Create custom profile on 1.18.2 Forge (any version).
3. Add CraftTweaker.
4. Launch.
5. Create new world.
6. Enter any of the abovementioned commands.

### Script used

No scripts used.

### The crafttweaker.log file

https://pastebin.com/wtY8HfKc

### Minecraft version

1.18

### Modloader

Forge

### Modloader version

40.1.84

### CraftTweaker version

9.1.197

### Other relevant information

Only JEI and JEITweaker installed.
No scripts.

### The latest.log file

https://pastebin.com/JtHA7Bbr

**Comment by jaredlll08** [comment]
Is there any output? like a "command not found"?

Can you run other commands like `/forge mods`?

**Comment by ShinQdan** [comment]
No, there is no output. Commands that I found working are dump, recipes and examples - they log to the file or create the examples but also no output is shown in the chat. Basically I do not have any output from CT in chat.
I can run other commands without problem.

I also updated to 1.18.2-9.1.197 but still got no chat output.

[crafttweaker.log](https://github.com/CraftTweaker/CraftTweaker/files/10099275/crafttweaker.log)
[latest.log](https://github.com/CraftTweaker/CraftTweaker/files/10099276/latest.log)

**Comment by jaredlll08** [comment]
Can you try and make a new modpack instance and see if you can still reproduce?

It is a very weird issue, no one else is reporting it, which leads me to believe that it must be something on your system.

**Comment by ShinQdan** [comment]
I tried using CourseForge app, MultiMC as well as installing Forge and CT manually with different forge versions (actually this was first thing I did before reporting the issue). All with the same result - none of those working. I am surprised by this too. I was even looking if there are some config or options which turn of the chat but found nothing.

Even more surprising is the fact that on 1.19.2 (Forge 43.1.47) on the same machine from the same CourseForge app CT is working perfectly fine and all the commands are working as expected.

**Comment by jaredlll08** [comment]
What language is your game set to use?

Does `/reload` give a "crafttweaker reload starting" message?

Can you also intentionally cause an error in a script file and see if that error is printed in chat?

Are you able to test commands from other mods and see if they output anything?

As far as I'm aware, we aren't doing anything out of the ordinary with how we are sending player messages, so it is very weird that forge commands are working but not CraftTweaker ones

**Comment by ShinQdan** [comment]
English (US)

No `/reload` does not give the message.

Yes, I can. Nothing happens when the script contains errors - no message in chat. Also the script without errors works fine.

Yes, I am able to use commands from other mods. I just installed MineColonies to the instance and its commands (e.g. `/mc help`, `/mc home`, `/mc rtp`, `/mc whoami`) are working fine and printing to chat.

**Comment by jaredlll08** [comment]
If you remove all your script files, like just rename the scripts folder completely, does you get any output from the commands?

**Comment by ShinQdan** [comment]
I tried removing scripts before but only deleting them, not renaming the folder, but the problem remained. Actually the problem had occurred before I even added any scripts. I have now renamed the scripts folder but commands are still not working.

**Comment by ShinQdan** [comment]
I just found that `/ct help` works and displays in chat properly.

Also I cloned your repo (the 1.18 branch) and launched the debug to try to find the problem but to my surprise it works perfectly fine. Therefore I tried different Forge and CT versions and even different Java distributions but nothing seems to fix the problem.

**Comment by ShinQdan** [comment]
I tried on a different machine and the problem persists... sort of... I downloaded CourseForge app on my friend's PC and launched 1.18.2 Forge with CT and it didn't work for me but when we logged into her account everything was working fine. I also tried different versions of MC and affected are versions 1.16 to 1.18. Other versions (1.19, 1.15 and down to 1.12) are working fine. Also other mods' commands are working on all versions that I checked.

This is very strange behavior. The fact that it is working on my friend's account suggests that it has something to do with my MS account. But at the same time other mods printing to chat without problem suggest it also has something to do with the CT itself. Also it is weird that only 3 major versions of MC are affected. I don't know much about what exactly changed in those versions and if it could have anything to do with the problem (I was stuck on 1.12.2 for a long time and just recently updated to 1.18.2) but it seems plausible it could.

I don't know if anything can be done to solve the problem but I'm curious what causes it. I will probably investigate further by myself but I don't really know where to start. If you've got any hints or suggestions please let me know.

**Comment by jaredlll08** [comment]
Have you tried on a server?
Are you able to test on fabric?

Looking at the code, in 1.18 I used the exact same method as forge to send messages for commands, I did change that in 1.19 (honestly not sure why I changed it actually), but regardless they end up calling the same code anyway to send it to the player.

I really wish I had an answer for you but I don't even know how I could start debugging this because it works fine for me and it seems to be specific to your account, maybe you somehow blocked the UUID that we sometimes use to send messages in 1.18 (which was removed in 1.19 and never existed in 1.12, but I am not sure about 1.15)

**Comment by ShinQdan** [comment]
I tested it on Fabric (v0.14.11, CT v9.1.177) and the commands are working.
I will later try it on a server.

Blocked the UUID? Never heard of something like this. And then why are commands from other mods working?

**Comment by BomberCost** [comment]
I am experiencing the same problem. One thing I found out was that it stopped working after I moved my pack from CurseForge to GDLauncher. Also I've searched for this damn UUID thingy, but I haven't find anything. All mods are working fine, configs file too, just the ct's chat commands don't show up in any way. Other thing may be java installation...but I don't see any possible cause to that.

Edit: I've seen also rendering errors. My skin wasn't showing (neither hand or entire pg in F5), items in hand or dropped not showing; entities spawned not rendering (they are there but invisible). I'm start thinking that my java is somehow broken.

**Comment by jaredlll08** [comment]
I am not sure about the UUID blocking, it was just something that I have heard potentially being a thing (being able to block a UUID).

@BomberCost I think your issue is very different than the original issue.




> I tested it on Fabric (v0.14.11, CT v9.1.177) and the commands are working. I will later try it on a server.
> 
> Blocked the UUID? Never heard of something like this. And then why are commands from other mods working?

If you somehow blocked the CraftTweaker UUID, is what I was talking about, but like I mentioned above, I'm not too sure about how it goes.

I am still not sure what is going on here, it doesn't make sense to work on the same computer but a different account, the only logical thing is that it is an account setting somehow.

**Comment by BomberCost** [comment]
Thanks for the clarification.

**Comment by jaredlll08** [comment]
I'm going to be closing this issue, unfortunately I don't even know where I would start debugging this without asking @ShinQdan to give me access to their account, which would be a very bad idea for them to do, and not something I would be super comfortable doing.

I really wish I had an answer to this one, but I don't believe we are doing anything different compared to forge and other mods.

**Comment by ShinQdan** [comment]
Well, debugging is no longer possible on my side either. I launched Minecraft today on the same instance I used when I found this bug and it magically works 🙃
I did not update or do anything with it. Actually I have not launched MC since November (except some dev forge builds). IDK why it suddenly started working. Maybe it was due to some changes I made to my Microsoft account recently but I have no way to confirm this.
